src/services/generateToken.ts

```diff
@@ -0,0 +1,51 @@
+const generateToken = async () => {
+
+    try {
+        // POST request using fetch with set headers
+        const urlencoded = new URLSearchParams();
+        urlencoded.append("grant_type", "client_credentials");
+        urlencoded.append("client_id", process.env.REACT_APP_SPOTIFY_CLIENT_ID || '' );
+        urlencoded.append("client_secret", process.env.REACT_APPP_SPOTIFY_CLIENT_SECRET || '');
+        const requestOptions = {
+            method: 'POST',
+            headers: {
+                "Content-Type": "application/x-www-form-urlencoded",
+            },
+            body: urlencoded
+        };
+        const data = await fetch('https://accounts.spotify.com/api/token', requestOptions)
+        const responseData = await data.json();
+        return responseData;
+    } catch (error) {
+        return;
```

Comment:
<a href="https://sourcery.ai"><img src="https://sourcery.ai/favicon.svg" height="15" alt="Sourcery logo" /></a>  **suggestion (code_refinement):** Returning `undefined` in case of an error during token generation might lead to unhandled exceptions later in the code where the token is used. It could be beneficial to throw a specific error or return a more descriptive error object to facilitate error handling and debugging.